src/main.js

```diff
@@ -114,60 +115,97 @@ class Game {
   }
 
   imageOnClick(idClicked) {
-    if (idClicked === this.prevClicked) return;
-    if (this.openedCards === 2) return;
-    this.cards[idClicked].open();
-    if (this.openedCards === 1) {
-      this.prevClicked = idClicked;
-      return;
+    if (this.fieldIsBlocked) return;
+    switch (this.openedCardsNum) {
+      case 0:
+        this.cards[idClicked].open();
+        return;
+      case 1:
+        if (this.isCardOpened(idClicked)) return;
+        this.cards[idClicked].open();
+        this.fieldIsBlocked = true;
+        this.checkCards();
+        break;
+      default:
     }
-    if (!this.cardsEqual(this.prevClicked, idClicked)) {
-      setTimeout(() => this.returnBackImage(idClicked), 1000);
+  }
+
+  isCardOpened(cardId) {
+    const openedCardId = parseInt(this.openedCards[0].div.id.slice(3), 10);
```

Comment:
That is something but I would like to split it even more.

Reply:
Ok, I'll split that at next review